### ISSUE TITLE
body disc indication

### ISSUE THREAD

**Issue by axd1967** [Stellarium/stellarium#1130]
**Is your feature request related to a problem? Please describe.**
In cases of very low illuminated sides, bodies (planets, moons) are barely visible. In case of a transit/occultation of a body by another body, it is very difficult to see when the occulted body becomes visible again.

**Describe the solution you'd like**
A toggle option to draw a circle that indicates the apparent disc This toggle should be independent of the "show/hide planets" option so that the user can choose whether to show bodies and/or discs.

Optionally, the disc would get a dotted/dashed appearance where (partially) hidden (so as to avoid color/night vision issues).

**Describe alternatives you've considered**
There is no alternative.

**Additional context**

Venus hding behind the Moon (2020-06-19) - works barely because of the small sickles.

![image](https://user-images.githubusercontent.com/3529789/84661057-0ec4a400-af1a-11ea-8019-a27f11c0c3e4.png)

but here, it is almost impossible to see Venus appearing behind the Moon

![image](https://user-images.githubusercontent.com/3529789/84661132-2f8cf980-af1a-11ea-99cd-c97775fca45d.png)

Imagine this is then transmitted to YouTube: it becomes very difficult to follow this phenomenon.
Once there is an option to draw circles for bodies, this is ok.

And if possible, this would be nice to have:
![image](https://user-images.githubusercontent.com/3529789/84661520-c659b600-af1a-11ea-8685-2e1d7659cc1b.png)

**Comment by gzotti** [comment]
We had complaints that showing the invisible edge of the planet/lunar disk, esp. in daylight, is "false", "misleading", "unrealistic". You cannot serve all... 

Grids: #485, but we need #502 before that to get the numbers right.

In any case, don't trust the times for Venus occultation, expect a minute of deviation or so. If you observe, please report GPS time with seconds, and your exact location, for "Venus hidden" and "Venus fully visible again". We can use such critical records for testing when aberration may be solved one day.

**Comment by axd1967** [comment]
yes, I don't expect sub-minute accuracy, I'm happy to live with this limitation, I understand this is part of Stellarium limits.

very good, thank you for suggesting to collect timings that can be used to check the model!

but sorry, those who complain about misleading should simply not use the toggle, let those that appreciate the feature use it. there is a clear use case where we need this, and technically it is very easy to do. or maybe make this an option that can only be set via config and keyboard, not via GUI?

**Comment by axd1967** [comment]
So, this is acceptable (and more realistic), but the disc that would replace the exact same dimensions would not be realistic? But both show exactly the same information?
![image](https://user-images.githubusercontent.com/3529789/86519086-b0635500-be37-11ea-939e-d9248adea95e.png)

**Comment by axd1967** [comment]
some freewheeling...

the astrocalc dialog uses this information:

https://github.com/Stellarium/stellarium/blob/da46a111638fbc3a963cd3a73956f7890267b963/src/gui/AstroCalcDialog.cpp#L4214-L4224

this could be used to compute the radius of the eclipsing body disc. admitted, (1) precision is not excellent (but again, the same problem happens when the user uses a fully illuminated eclipsing body instead - this is also wrong, but this cannot be prevented) and (2) a spheroid should not be drawn as a circle, and (3) eclipsed bodies might also need to be drawn correctly (correct angular size, with a minimum of x pixels, in order to stay visible).
but it is a first approximation in order to go forward in this issue: to give some indication where the eclipse ends in case this happens on the dark side. but this does not solve the issue completely: the eclipsing body must be replaced by the body circle because the idea is to see _where_ the eclipsed object is behind the eclipsing object.

an option "discs only" (tooltip: "_this replaces all solar system bodies with circles in order to better see eclipse start/endings_") could be added in the View/SSO tab + keyboard shortcut to easily toggle the option.

eclipsing bodies can be Sun, Moon, planet.

(the hint circle is unusable, in fact a texture:)
https://github.com/Stellarium/stellarium/blob/da46a111638fbc3a963cd3a73956f7890267b963/src/core/modules/SolarSystem.cpp#L269
![image](https://github.com/Stellarium/stellarium/blob/da46a111638fbc3a963cd3a73956f7890267b963/textures/planet-indicator.png)

**Comment by axd1967** [comment]
another example where a body disc is handy

![image](https://user-images.githubusercontent.com/3529789/94931755-6a666d80-04c8-11eb-9932-8b8b12fc2d98.png)

by lack of a disc, a telrad ring is used (but one cannot see what is happening behind the Moon disc).

(the text hint is missing, the Moon is just too close for it to appear)

**Comment by axd1967** [comment]
compare with #430: a disc that would draw the outline of the current Moon (simulated here by the Moon simulating the eclipsed object , and the umbra simulating the Moon...)

![image](https://user-images.githubusercontent.com/3529789/108914823-3b45c980-762c-11eb-8522-49a645ed92e0.png)

![image](https://user-images.githubusercontent.com/3529789/108914884-57496b00-762c-11eb-9558-ae18bbff17e1.png)

**Comment by axd1967** [comment]
Example of Moon occulting Saturn:

<img width="1050" height="576" alt="Image" src="https://github.com/user-attachments/assets/ac47dbae-d250-4457-9fd1-92b92920fc44" />

Source: https://iota-es.de/moon/moon_occ_saturn_20250104.html
